### DISCUSSION TITLE
Feature request: Allow Custom Icon Layers to be Named and Added to the Library.

### DISCUSSION THREAD

**Discussion by Kenny-Moore** [FortAwesome/Font-Awesome#20770] [Feature Requests]
### Layers are very powerful yet not easily reusable. 

When configuring the Fontawesome 5 library you must import and add the icons you wish to use; why not also have the ability to define custom icon layers that you would be able to name, add and use in the app like any other normal icon. 

With this enhancement we should be able to write something like this:
```javascript
import fontawesome from '@fortawesome/fontawesome';
import faSpinner from '@fortawesome/fontawesome-pro-regular/faSpinner';

let faCustomLayer = fontawesome.layer((push) => {
  push(fontawesome.icon(faSpinner))
  push(fontawesome.text('Wait…', { transform: { size: 4 } } ))
})

fontawesome.library.addLayer({iconName: 'wait', icon: faCustomLayer}); // or something similar
```

And be able to get the saved layer like this:

```javascript
let faWaitLayer = fontawesome.icon({ prefix: 'fa-layers', iconName: 'wait' });
```
### or in the HTML with:

```html
<i class="fa-layers fa-layer-wait"></i>
```

### Saved layers should behave exactly like their native counterparts.
Concider this example, using the vueJS binding. The developer here has mapped an array of actions to specific icon names and prefixes and is using a for loop binding to render their actions:

```html
<ul id="nav-1">
  <li v-for="action in actions">
    <font-awesome-icon :icon="action.icon" /> {{ action.label }}
  </li>
</ul>
```

It would be difficult for this developer to use a icon layer for one of those actions if their designer required it. How would you suggest solving this problem the way the framework is currently written?

By allowing Layers to be registered as "custom icons" that work like any other icon you would allow developers to use the same strategy to display icons and layers when necessary.

Please consider doing this sooner rather than later.

**Comment by sensibleworld** [comment]
Doing a little management in Github and (batch) closing out potentially stale issues. If the above still needs attention, please let us know and we’ll re-open!

**Comment by shaedrich** [comment]
Has it only been closed because the FontAwesome team did not reply to this until now? If that's the case, that's a very poor explanation.

**Comment by sensibleworld** [comment]
@shaedrich As the note above mentions, we’ve been doing some batch cleanup of older issues. Our GitHub repo has gotten pretty unwieldy with the number of open threads, and we’re trying to bring things back to a manageable level (including possibly moving icon requests elsewhere). Right now it’s hard to see the trees for the forest.

Since this request still has interest, I’m happy to re-open it. Apologies for any frustration this may hav caused.

**Reply by sensibleworld** [reply]
Also moved this to a discussion, as opposed to an issue.

**Reply by shaedrich** [reply]
Thanks 👍🏻
